src/api/persistance-store.model.ts

```diff
@@ -2,49 +2,77 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 import { ThunkDispatch } from "redux-thunk";
-import { ScheduleDataModel } from "../common-models/schedule-data.model";
+import { MonthDataModel, ScheduleDataModel } from "../common-models/schedule-data.model";
 import { WorkerInfoModel } from "../common-models/worker-info.model";
 import { ActionModel } from "../state/models/action.model";
 import { ApplicationStateModel } from "../state/models/application-state.model";
-
-/*eslint-disable @typescript-eslint/camelcase */
+import { ShiftInfoModel } from "../common-models/shift-info.model";
 
 export type ThunkFunction<T> = (
   dispatch: ThunkDispatch<ApplicationStateModel, void, ActionModel<T>>,
   getState: () => ApplicationStateModel
-) => Promise<unknown>;
+) => Promise<unknown> | unknown;
+
+export type ScheduleKeyString = string;
+export type UpdatePosition = "HEAD" | "TAIL";
+
+export class ScheduleKey {
+  constructor(public month: number, public year: number) {}
+
+  get dbKey(): ScheduleKeyString {
+    return `${this.month}_${this.year}`;
+  }
+
+  get nextMonthKey(): ScheduleKey {
+    const isLastYearMonth = this.month === 11;
+    const month = isLastYearMonth ? 0 : this.month + 1;
+    const year = isLastYearMonth ? this.year + 1 : this.year;
+    return new ScheduleKey(month, year);
+  }
 
-export interface ScheduleKey {
-  month: number;
-  year: number;
+  get prevMonthKey(): ScheduleKey {
+    const isFirstYearMonth = this.month === 0;
+    const month = isFirstYearMonth ? 11 : this.month - 1;
+    const year = isFirstYearMonth ? this.year - 1 : this.year;
+    return new ScheduleKey(month, year);
+  }
```

Comment:
ten kod różni się tylko 0/11 (borderMonth) i +1 / -1 (increment)

Reply:
może trochę czepianie się bo to tylko 4 linijki xd

Reply:
zrób jak uważasz, możesz zostwić albo przerobić

Reply:
Faktycznie, zrobię to jakoś ładniej, żeby nie duplikować :+1: